src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -504,8 +527,19 @@ public static Pair<ConnectionString, DriverPropertyInfo[]> getConnectionSettings
         }
 
         ParseResult result = normalizeConnectionOptions(originalConnectionString, info);
-        String user = result.user;
-        char[] password = result.password;
+        String user = null;
+
+        char[] password = null;
+        char[] x509Passphrase = null;
+
+        if (MONGODB_X509.equalsIgnoreCase(result.authMechanism)) {
```

Comment:
[note]
Since I played with this recently, I noticed that we can call result.getAuthenticationMechanism and use the Enum AuthenticationMechanism. Not relying on string values is a bit better. More built-in safety and simplified check.
If you feel like it, feel free to make the change on this PR. If not, that's alright too,

Reply:
That works, I made the change.

Reply:
Thank you :)